### DISCUSSION TITLE
How to know if an NPM package support gulp 4?

### DISCUSSION THREAD

**Discussion by moatefdev** [gulpjs/gulp#2539] [General] [answered]
I sometimes found that some packages do not work, and the reason for not working is that this package **does not support Gulp version 4.** So, how to know if an NPM package **support gulp 4**?

**Comment by phated** [comment, marked as answer]
You can't really. If a plugin was written **correctly** for gulp 3, then it would work with gulp 4. I assume you are trying to grab things that didn't follow the guidelines.

However, gulp 4 was designed in a way that any JS code can be used (not just gulpplugins), so you can easily re-implement some behavior you are looking for.
